urlview: drop

Ma27 opened the pull request:
## Description of changes

There was a single occurence, `tmuxPlugins.urlview`. This also supports an urlview alternative (`extract_url`), so I changed it to use that instead of `urlview`.

---

Closes #273406

There are no upstream maintainers anymore[1]. This tool is known to
wrongly truncate URLs (this broke invite links a few times for me).

Now it's officially dead and there are a few useful alternatives,
`urlscan` & `extract_url`.

[1] https://salsa.debian.org/debian/urlview/-/commit/4ad7d236c745b665db1d2fd07ce5ffc1748e4240

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Ma27 committed:
tmuxPlugins.urlview: switch to extract_url

The tool supports both programs, however `urlview` will be removed.

Ma27 committed:
urlview: remove

Closes #273406

There are no upstream maintainers anymore[1]. This tool is known to
wrongly truncate URLs (this broke invite links a few times for me).

Now it's officially dead and there are a few useful alternatives,
`urlscan` & `extract_url`.

[1] https://salsa.debian.org/debian/urlview/-/commit/4ad7d236c745b665db1d2fd07ce5ffc1748e4240

Ma27 commented:
cc @maxbrunet

SuperSandro2000 commented on pkgs/top-level/aliases.nix:
```diff
@@ -963,6 +963,10 @@ mapAliases ({
   unifi-poller = unpoller; # Added 2022-11-24
   unifiStable = unifi6; # Added 2020-12-28
   untrunc = untrunc-anthwlock; # Added 2021-02-01
+  urlview = throw ''
+    'urlview' has been dropped because it's unmaintained.
+    Consider switching to an alternative such as `pkgs.extract_url` or `pkgs.urlscan`.
+  ''; # Added 2023-12-14
```
```suggestion
  urlview = throw "'urlview' has been dropped because it's unmaintained. Consider switching to an alternative such as `pkgs.extract_url` or `pkgs.urlscan`."; # Added 2023-12-14
```
no idea if that would break the scripts scanning this file, so better safe than sorry

Ma27 replied:
I think it's a very bad idea to do that, especially without a very clear warning sign (yes, a maintainer script is referenced, but it doesn't tell me how it deals with this file and I won't dig into that now ;-) )

Ma27 committed:
aliases: convert `urlview` throw to a oneliner

I really, really hope that nobody actually tries to parse Nix code and blindly relies on how it's structured though.

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>
